syntax = "proto3";
option cc_enable_arenas = true;

package YandexQuery.Internal;
option java_package = "com.yandex.query.internal";
option java_outer_classname = "YandexQueryInternalProtos";

import "ydb/library/yql/providers/dq/api/protos/service.proto";
import "ydb/library/yql/dq/proto/dq_tasks.proto";
import "ydb/public/api/protos/yq.proto";
import "ydb/core/yq/libs/protos/yq_private.proto";

import "google/protobuf/duration.proto";

message Consumer {
    string name = 1;
}

message QueryInternal {
    string token = 1;
    repeated NYql.NDqProto.TDqTask task = 2; // deprecated and should not be used, will be removed in future
    repeated Yql.DqsProto.TAttr settings = 3;
    repeated Consumer consumer = 4;
    repeated YandexQuery.Connection connection = 5;
    repeated YandexQuery.Binding binding = 6;
    YandexQuery.QueryAction action = 7;
    string ast = 8; // deprecated and should not be used, will be removed in future
    ExecuteMode execute_mode = 9;
    StateLoadMode state_load_mode = 10;
    string cloud_id = 11;
    repeated Yq.Private.TopicConsumer created_topic_consumers = 12;
    repeated bytes dq_graph = 13; // deprecated: use dq_graph_compressed
    int32 dq_graph_index = 14;
    StreamingDisposition disposition = 15;
    uint64 result_limit = 16;
    google.protobuf.Duration execution_ttl = 17;
    Yq.Private.CompressedData ast_compressed = 18;
    Yq.Private.CompressedData plan_compressed = 19;
    repeated Yq.Private.CompressedData dq_graph_compressed = 20;
}

message JobInternal {
    Yq.Private.CompressedData ast_compressed = 1;
    Yq.Private.CompressedData plan_compressed = 2;
    repeated Yq.Private.CompressedData dq_graph_compressed = 3;
}

message ConnectionInternal {
    string cloud_id = 2;
}

message BindingInternal {
    string cloud_id = 2;
}
